### PR TITLE
Issue 1392 by Pjohans: Make a checkbox to activate filtering on libra…

### PR DESCRIPTION
**Pull request by pjohans**
…ries

### CODE REVIEW

File: ting.admin.inc [hunk 1]
```diff
@@ -35,6 +35,14 @@ function ting_admin_ting_settings($form_state) {
     '#default_value' => variable_get('ting_search_url', ''),
   );
 
+  $form['ting']['ting_search_filter_holdingsitem']= array(
+    '#type' => 'checkbox',
+    '#title' => t('Filter search queries'),
+    '#description' => t('Filter searches by holdingsitem.agencyId. Only activate when opensearch uses data-well 3.5, and the provider is fbs'),
+    '#required' => FALSE,
+    '#default_value' => variable_get('ting_search_filter_holdingsitem',0),
```
Comment (cableman) [inline ting.admin.inc]: Missing space (variable_get('ting_search_filter_holdingsitem', 0)) before 0.

File: ting.admin.inc [hunk 2]
```diff
@@ -35,6 +35,14 @@ function ting_admin_ting_settings($form_state) {
     '#default_value' => variable_get('ting_search_url', ''),
   );
 
+  $form['ting']['ting_search_filter_holdingsitem']= array(
+    '#type' => 'checkbox',
+    '#title' => t('Filter search queries'),
```
Comment (kasperg) [inline ting.admin.inc]: Perhaps the title could be improved a bit. This seems a little generic.

**Comment by kasperg** [comment]
Review complete. A bit of nitpicking. Otherwise ready for merging.

**Comment by xendk** [comment]
Here's what we agreed on, on skype:

@pjohans 
Remove the "$query = _ting_search_quote($query);" (and the corresponding quoting on https://github.com/ding2/ting_search/pull/31/files if that should still be merged).

Rename the variable from `ting_search_filter_holdingsitem` to `ting_filter_by_local_holdings`

About here: https://github.com/ding2/ting/blob/master/ting.client.inc#L228 to https://github.com/ding2/ting/blob/master/ting.client.inc#L233 replace the code with:

```php
  $request = ting_get_request_factory()->getSearchRequest();
  $agency = variable_get('ting_agency', FALSE);

  if ($agency && variable_get('ting_filter_by_local_holdings', 0)) {
    // Limit the search to materials from the local library. From well 3.5 each
    // library is no longer isolated.
    $query = $query . '(' . $query . ') and holdingsitem.agencyid="' . $agency . '"';
  }
  
  $request->setQuery($query);
  if ($agency) {
    $request->setAgency($agency);
  }
  $request->setStart($results_per_page * ($page - 1) + 1);
```

That should be it. Options for bypassing the filter can be added later.

File: ting.client.inc [hunk 3]
```diff
@@ -226,6 +226,16 @@ function ting_get_objects($ids) {
  */
 function ting_do_search($query, $page = 1, $results_per_page = 10, $options = array()) {
   $request = ting_get_request_factory()->getSearchRequest();
+
+  $query = _ting_search_quote($query);
```
Comment (kasperg) [inline ting.client.inc]: As mentioned by @xendk `_ting_search_quote` should not be called here. Please remove this line. Here we expect the query to be valid.

File: ting.client.inc [hunk 4]
```diff
@@ -226,6 +226,16 @@ function ting_get_objects($ids) {
...
+
+  $agency = variable_get('ting_agency', FALSE);
+  if ($agency && variable_get('ting_filter_by_local_holdings', 0)) {
+    // Limit the search to materials from the local library. From well 3.5 each
+    // library is no longer isolated.
```
Comment (kasperg) [inline ting.client.inc]: :+1: Good commenting.

**Comment by kasperg** [comment]
Review complete.

**Comment by cableman** [comment]
@kasperg Remember to update https://github.com/kasperg/ding2/commits/consolidation as well.

**Pull request merged**